src/app/dev/DevToys.Business/ViewModels/MainWindowViewModel.cs

```diff
@@ -154,6 +172,71 @@ internal void UpdateWindowTitle(bool isInCompactOverlayMode)
         }
     }
 
+    /// <summary>
+    /// Get the data from the clipboard and try to detect tools that can accept the clipboard data as an input.
+    /// </summary>
+    internal async Task RunSmartDetectionAsync(bool isInCompactOverlayMode)
+    {
```

Comment:
Something I should improve here:

Smart Detection happens in a background thread and doesn't freeze the UI anymore. That's great! However it may still take 5-10 seconds, and there's no indication of what's going on, and Smart Detection may navigate the user automatically to a tool we detected when done.

This can be troublesome for the user experience. Here are things we could do to improve the UX (but let's do that in a separated PR?) :

1. Set a timeout so we don't spend more than 2-3 seconds doing the detection.
2. if user navigated somewhere by himself since we started the detection, cancel the detection, because it sounds like the user knowns where he's going.

Thoughts @btiteux  ?

Reply:
Done